lib/connectors/base/simple_rules_parser.rb

```diff
@@ -11,23 +11,144 @@
 
 module Connectors
   module Base
+    class FilteringRulesValidationError < StandardError; end
+
     class SimpleRulesParser
+
+      include Core::Filtering
+
+      attr_reader :rules
+
       def initialize(rules)
-        @rules = (rules || []).map(&:with_indifferent_access).filter { |r| r[:id] != 'DEFAULT' }.sort_by { |r| r[:order] }
+        begin
+          sorted = (rules || []).map { |r| SimpleRule.new(r) }.filter { |r| r.id != 'DEFAULT' }.sort_by(&:order)
```

Comment:
I like this change. I'd avoided it because of the `order` requirement, but you've solved that now. 
Should we unify this sorting with the one done by the PostProcessingEngine to DRY up our logic?